sympy/series/gruntz.py

```diff
@@ -416,19 +416,19 @@ def limitinf(e, x, leadsimp=False):
     ``e`` cannot be simplified.
     """
     # rewrite e in terms of tractable functions only
-    e = e.rewrite('tractable', deep=True)
 
     if not e.has(x):
         return e  # e is a constant
     if e.has(Order):
         e = e.expand().removeO()
-    if not x.is_positive:
-        # We make sure that x.is_positive is True so we
-        # get all the correct mathematical behavior from the expression.
+    if not x.is_positive or x.is_integer:
+        # We make sure that x.is_positive is True and x.is_integer is False
+        # so we get all the correct mathematical behavior from the expression.
         # We need a fresh variable.
         p = Dummy('p', positive=True, finite=True)
```

Comment:
`finite` is redundant here when used with `positive`

Reply:
Alright will just see if it does not break anything. If it doesn't will make the commit.